nvme-print-json.c

```diff
@@ -2164,14 +2203,34 @@ static void json_phy_rx_eom_descs(struct nvme_phy_rx_eom_log *log,
 		obj_add_uint(jdesc, "bottom", le16_to_cpu(desc->bottom));
 		obj_add_uint(jdesc, "left", le16_to_cpu(desc->left));
 		obj_add_uint(jdesc, "right", le16_to_cpu(desc->right));
-		obj_add_uint(jdesc, "nrows", le16_to_cpu(desc->nrows));
-		obj_add_uint(jdesc, "ncols", le16_to_cpu(desc->ncols));
-		obj_add_uint(jdesc, "edlen", le16_to_cpu(desc->edlen));
+		obj_add_uint(jdesc, "nrows", nrows);
+		obj_add_uint(jdesc, "ncols", ncols);
+		obj_add_uint(jdesc, "edlen", edlen);
 
 		if (NVME_EOM_ODP_PEFP(log->odp))
-			allocated_eyes[i] = json_eom_printable_eye(desc, r);
+			allocated_eyes[i] = json_eom_printable_eye(desc, jdesc);
+
+		if (edlen == 0)
+			continue;
+
+		/* 2 hex chars + space per byte */
+		_cleanup_free_ char *hexstr = malloc(edlen * 3 + 1);
```

Comment:
please avoid adding definition inside the code block. add them at the begin of the function or the code block.